test/unit/xcode-specs.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+var fs = require('fs')
+  , Q = require('q')
+  , xcode = require('../../lib/devices/ios/xcode.js')
+  , chai = require('chai')
+  , chaiAsPromised = require('chai-as-promised');
+
+chai.should();
+chai.use(chaiAsPromised);
+
+var fileExists = Q.denodeify(fs.exists);
+
+describe('xcode.js @skip-linux', function () {
+
+  it('should find the automation trace template', function (done) {
+    xcode.getAutomationTraceTemplatePath()
+    .then(function (traceTemplatePath) {
+      fileExists(traceTemplatePath).should.eventually.be.true;
+      var suffix = ".tracetemplate";
+      traceTemplatePath.slice(-suffix.length).should.equal(suffix);
+    })
+    .nodeify(done);
+  });
+
```

Comment:
can we add a test that we get an error when it can't find the tracetemplate? and in both these cases we could just mock xcode.getPath to return without actually needing to call xcode-select